mllib/src/main/scala/org/apache/spark/ml/tree/impl/GradientBoostedTrees.scala

```diff
@@ -299,26 +317,25 @@ private[spark] object GradientBoostedTrees extends Logging {
     baseLearners(0) = firstTreeModel
     baseLearnerWeights(0) = firstTreeWeight
 
-    var predError: RDD[(Double, Double)] =
-      computeInitialPredictionAndError(input, firstTreeWeight, firstTreeModel, loss)
+    var predError = computeInitialPredictionAndError(input, firstTreeWeight, firstTreeModel, loss)
     predErrorCheckpointer.update(predError)
```

Comment:
if we are going to keep the checkpointing of unweighted error as opposed to weighted error, then it would be nice to specify that in the name of the checkpointer:
predUnweigtedErrorCheckpointer
or alternatively add a comment to make that clear:
```
    // Note: this is checkpointing the unweighted error
    predErrorCheckpointer.update(predError) 
```